### DISCUSSION TITLE
Accept() returns -22

### DISCUSSION THREAD

**Discussion by yusufKilicarslan** [ANLAB-KAIST/KENSv3#51] [Project Q&A 1. Application Layer (Socket)] [answered]
Why does this happen?
I thought socketfd must be nonnegative.

**Comment by Pusnow** [comment, marked as answer]
22 (EINVAL) means

> Socket is not listening for connections, or addrlen is
>              invalid (e.g., is negative).


https://man7.org/linux/man-pages/man2/accept.2.html

**Reply by bongjunj** [reply]
Why the return value, instead of `errno`, is used to indicate an error?
